### PR TITLE
feat: add semantic release

### PR DESCRIPTION
**Pull request by Kininaru**

### CODE REVIEW

**Comment by Kininaru** [comment]
@hsluoyz @Zxilly @sh1luo @WindSpiritSR @1340908470 Please review.

**Comment by hsluoyz** [comment]
@Kininaru did you try it in your repo?

File: .github/workflows/release.yml [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+    branches:
+      - master
+jobs:
+  semantic-release:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2
+
+      - name: Run semantic-release
+        if: github.repository == 'casbin/node-casbin' && github.event_name == 'push'
```
Comment (Zxilly) [inline .github/workflows/release.yml:13]: wrong repo name

File: .github/workflows/release.yml [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+        run: yarn install && yarn semantic-release
```
Comment (Zxilly) [inline .github/workflows/release.yml:14]: `semantic-release` script didn't exist.

File: .github/workflows/release.yml [hunk 3]
```diff
@@ -0,0 +1,17 @@
+name: Release
```
Comment (Zxilly) [inline .github/workflows/release.yml:1]: And plz also add `.releaserc.json`

**Review by Zxilly — CHANGES_REQUESTED** [review]

**Comment by sh1luo** [comment]
@Kininaru I think you can refer to the implementation of casdoor, these two requirements are basically the same.

**Comment by hsluoyz** [comment]
@Zxilly

**Review by Zxilly — APPROVED** [review]

File: .github/workflows/release.yml [hunk 4]
```diff
@@ -0,0 +1,16 @@
...
+jobs:
+  semantic-release:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2
+
+      - name: Run semantic-release
+        run: yarn install && yarn global add semantic-release@17.4.4 && semantic-release
+        env:
+          GITHUB_TOKEN: ${{ secrets.GH_TOKEN }}
```
Comment (Zxilly) [inline .github/workflows/release.yml:15]: plz use `secrets.GITHUB_TOKEN`, which is auto-generated by GitHub Actions

**Review by Zxilly — CHANGES_REQUESTED** [review]

**Comment by Zxilly** [comment]
@hsluoyz plz set up `NPM_TOKEN` in this org.

**Comment by hsluoyz** [comment]
> NPM_TOKEN

done:

image

**Comment by Kininaru** [comment]
@hsluoyz @Zxilly Please review.

**Review by Zxilly — APPROVED** [review]

**Pull request merged**